sonic_platform_base/sonic_xcvr/api/public/c_cmis.py

```diff
@@ -257,8 +260,31 @@ def get_pm_all(self):
         PM_dict['rx_mer_avg'] = self.xcvr_eeprom.read(consts.RX_AVG_MER_PM)
         PM_dict['rx_mer_min'] = self.xcvr_eeprom.read(consts.RX_MIN_MER_PM)
         PM_dict['rx_mer_max'] = self.xcvr_eeprom.read(consts.RX_MAX_MER_PM)
+
+        self.xcvr_eeprom.write(consts.VDM_CONTROL, VDM_UNFREEZE)
+        #Retry for freeze done status
+        if self.util_vdm_freeze_unfreeze_status_retry(VDM_FREEZE_UNFREEZE_DONE_RETRY, SLEEP_50MS, consts.VDM_UNFREEZE_DONE, 1):
+            helper_logger.log_error('UnFreeze done status check timeout expired')
+
         return PM_dict
 
+    def util_vdm_freeze_unfreeze_status_retry(self, retry, sleep_time, reg_field, expected_status):
+        '''
+         Utility function which retries for expected status with timeout and sleep between retries.
+         Input args: retry count, sleep time between retries, register to read for status, expected status to compare.
+         output : 0 - register read mactches expected status, 1 - retry timed out.
+         '''
+        time.sleep(sleep_time)
+        status = 0
+        while(retry > 0):
+            status = self.xcvr_eeprom.read(reg_field)
+            if (status == expected_status) or (retry <= 0):
+                return 0
+            retry = retry-1
+            time.sleep(sleep_time)
+        if (retry <= 0) and (status != expected_status):
+            return 1
+
```

Comment:
@jaganbal-a can we have the retry logic inside the PM thread instead of in the API. 
Also module may take at max tVDMF seconds to freeze/unfreeze. see tVDMF

<img width="771" alt="image" src="https://github.com/sonic-net/sonic-platform-common/assets/45705344/5ad82158-1774-43a2-98ee-f70af949ed9e">